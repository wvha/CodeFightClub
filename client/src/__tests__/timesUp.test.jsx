import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import TimesUp from '../components/timesUp.jsx';

Enzyme.configure({ adapter: new Adapter() });

test('timesUp should contain \'GAME OVER \'', () => {
  const component = shallow(<TimesUp changeView={() => {}}/>);
  expect(component.contains('GAME OVER')).toBe(true);
})

test('timesUp should contain \'PLAY AGAIN \'', () => {
  const component = shallow(<TimesUp changeView={() => {}}/>);
  expect(component.contains('PLAY AGAIN')).toBe(true);
})

test('timesUp should render properly', () => {
  const component = toJson(shallow(<TimesUp changeView={() => {}}/>));
  expect(component).toMatchSnapshot();
})

// test('timesUp should render properly', () => {
//   const component = shallow(<TimesUp changeView={() => {}} gameTimer={5} view={'gameRoom'}/>);
//   console.log(component.find('ReactModal'));
//   expect(component.find('ReactModal').props().isOpen).toBe(false);
// })