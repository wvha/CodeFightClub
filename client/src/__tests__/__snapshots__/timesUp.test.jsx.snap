// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`timesUp should render properly 1`] = `
<span>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="Modal container"
    closeTimeoutMS={0}
    contentLabel="SignUp Modal"
    isOpen={false}
    overlayClassName="Overlay"
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <span>
      GAME OVER
    </span>
    <br />
    <br />
    <button>
      PLAY AGAIN
    </button>
  </Modal>
</span>
`;
